base/abstractarray.jl

```diff
@@ -323,6 +323,16 @@ function _all_match_first(f::F, inds, A, B...) where F<:Function
 end
 _all_match_first(f::F, inds) where F<:Function = true
 
+"""
+    eachstoredindex(A)
+
+Returns an iterable over the indices of `A` where the values are structurally non-zero.
```

Comment:
Do we want to define this as being tied to structural **nonzeros**?  Or simply stored values?  We could simultaneously introduce a `unstoredvalue` to give room for other sorts of structures.

In any case, I like this and think it's very needed.

Reply:
Good question. What would be expected for a `FillArrays.Ones`? In the case of stored values, it would return an empty iterator? I am not sure we can make this useful in general, except for constructors?